src/platforms/shared/tracking/batch-processor.ts

```diff
@@ -0,0 +1,32 @@
+export class BatchProcessor {
+	batchedTasks = [];
+
+	constructor(private tasksList: any[], private batchSize: number, private delay: number) {
+		this.batchTasks();
+	}
+
+	private batchTasks(): void {
+		this.batchedTasks = [...this.tasksList].reduce((resultArray, item, index) => {
+			const chunkIndex = Math.floor(index / this.batchSize);
+
+			if (!resultArray[chunkIndex]) {
+				resultArray[chunkIndex] = [];
+			}
+
+			resultArray[chunkIndex].push(item);
+
+			return resultArray;
+		}, []);
+	}
+
+	dispatchEventsWithTimeout(sendRequest): void {
+		const batchTimer = setInterval(() => {
+			if (this.batchedTasks.length > 0) {
```

Comment:
quick exit should be more readable, hm?
```suggestion
			if (this.batchedTasks.length <= 0) { clear; return; }
```

Reply:
👍